src/h5web/visualizations/nexus/utils.test.ts

```diff
@@ -0,0 +1,330 @@
+    it('should return `true` with known NeXus interpretation', () => {
+      expect(isNxInterpretation('spectrum')).toBe(true);
+    });
+
+    it('should return `false` with invalid or unknown NeXus interpretation', () => {
+      expect(isNxInterpretation(42)).toBe(false);
+      expect(isNxInterpretation('unknown')).toBe(false);
+    });
+  });
+
+  describe('getNxAxes', () => {
+    it('should return array of axes', () => {
+      const axesAttr = makeNxAxesAttr(['X', 'Y']);
+      const group = makeGroup('foo', [], { attributes: [axesAttr] });
+
+      expect(getNxAxes(group)).toEqual(['X', 'Y']);
+    });
+
+    it('should wrap single axis string in array', () => {
+      const axesAttr = makeStrAttr('axes', 'X');
+      const group = makeGroup('foo', [], { attributes: [axesAttr] });
+
+      expect(getNxAxes(group)).toEqual(['X']);
+    });
+
+    it('should return empty array if group has no `axes` attribute', () => {
+      const group = makeGroup('foo');
+      expect(getNxAxes(group)).toEqual([]);
+    });
+
+    it('should throw if `axes` value is neither string nor array', () => {
+      const axesAttr = makeScalarIntAttr('axes', 42);
+      const group = makeGroup('foo', [], { attributes: [axesAttr] });
+
+      expect(() => getNxAxes(group)).toThrow(/array/u);
+    });
+  });
+
+  describe('getDatasetLabel', () => {
+    it('should return value of `long_name` attribute in priority if available', () => {
+      const dataset = makeDataset('foo', scalarShape, intType, {
+        attributes: [
+          makeStrAttr('long_name', 'Foobar'),
+          makeStrAttr('units', 'nm'),
+        ],
+      });
+
+      expect(getDatasetLabel(dataset)).toBe('Foobar');
+    });
+
+    it('should combine dataset name and value of `units` attribute if available', () => {
+      const dataset = makeDataset('foo', scalarShape, intType, {
+        attributes: [makeStrAttr('units', 'nm')],
+      });
+
+      expect(getDatasetLabel(dataset)).toBe('foo (nm)');
+    });
+
+    it('should fall back to dataset name', () => {
+      const dataset = makeDataset('foo', scalarShape, intType, {
+        attributes: [
+          makeScalarIntAttr('long_name', 42),
+          makeScalarIntAttr('units', 42),
+        ],
+      });
+
+      expect(getDatasetLabel(dataset)).toBe('foo');
+    });
+  });
+
+  describe('getSilxStyle', () => {
+    it('should parse `SILX_style` attribute', () => {
+      const silxStyle = {
+        signalScaleType: ScaleType.SymLog,
+        axesScaleType: [ScaleType.Log],
+      };
+
+      const group = makeGroup('foo', [], {
+        attributes: [makeSilxStyleAttr(silxStyle)],
+      });
+
+      expect(getSilxStyle(group)).toEqual(silxStyle);
+    });
+
+    it('should ignore unknown `SILX_style` options and invalid values', () => {
+      const silxStyle1 = {
+        unknown: ScaleType.Log,
+        signal_scale_type: 'invalid',
+        axes_scale_type: ScaleType.Linear, // should be array
```

Comment:
For consistency with `axes`, should it not convert this as an array ?

Also, I would perhaps add a check in `useNxData` to see if `axes` and `axes_scale_type` have the same length, logging a warning if not.

Reply:
👍 I've used `should fall back to dataset name when other attributes are absent or not string` and added a check for a dataset with no attributes.

Reply:
Totally agree for the first part.

For the second part, I've had a play with it and it doesn't seem worth it, especially since if the lengths don't match, nothing breaks -- you just don't get the scale types you were hoping for. If `SILX_style` was really important, then it'd make sense to check it, but just logging a warning is not very helpful: the user would still have to realise that something is not right.

Reply:
>  you just don't get the scale types you were hoping for.
That what I expected. I thought it might be worth a warning as it might not be obvious to debug. 

But in the end, the user can still change the scales themselves so I won't push strongly for this.